shell/packages/sandstorm-ui-powerbox/powerbox-client.js

```diff
@@ -193,133 +133,103 @@ SandstormPowerboxRequest = class SandstormPowerboxRequest {
     });
   }
 
-  selectApiToken(apiTokenCard) {
-    Meteor.call("getViewInfoForApiToken", apiTokenCard._id, (err, result) => {
-      if (err) {
-        console.log(err);
-        this._error.set(err.toString());
-      } else {
-        const viewInfo = result;
-        this.annotateViewInfo(viewInfo);
-        this._selectedProvider.set({
-          type: "frontendref-uiview",
-          grainId: apiTokenCard.grainId,
-          apiToken: apiTokenCard._id,
-          title: apiTokenCard.title,
-          templateName: "powerboxProviderUiView",
-          templateData: () => {
-            return {
-              _id: apiTokenCard._id,
-              title: apiTokenCard.title,
-              appTitle: apiTokenCard.appTitle,
-              iconSrc: apiTokenCard.iconSrc,
-              lastUsed: apiTokenCard.lastUsed,
-              viewInfo: viewInfo,
-              onComplete: (roleAssignment) => {
-                this.completeUiView(roleAssignment);
-              },
-            };
-          },
-        });
+  filteredCardData() {
+    const cards = PowerboxOptions.find({ requestId: this._requestId }).map(cardData => {
+      // Use ID as title if we don't find anything better.
+      cardData.title = cardData._id;
+
+      if (cardData.grainId) {
+        this.extendWithGrainInfo(cardData);
```

Comment:
Added comments. Refactoring into a class might be a good idea but not going to do it right now.